sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/ChatCompletionsOptions.java

```diff
@@ -529,12 +554,10 @@ public FunctionCallConfig getFunctionCall() {
     public ChatCompletionsOptions setFunctionCall(FunctionCallConfig functionCallConfig) {
         this.functionCallConfig = functionCallConfig;
         if (FunctionCallPreset.values().stream()
-                .anyMatch(preset -> preset.toString().equals(functionCallConfig.getName()))) {
-            this.functionCall =
-                    new FunctionCallPresetFunctionCallModel(
-                            FunctionCallPreset.fromString(this.functionCallConfig.getName()));
+            .anyMatch(preset -> preset.toString().equals(functionCallConfig.getName()))) {
+            this.functionCall = BinaryData.fromObject(FunctionCallPreset.fromString(this.functionCallConfig.getName()));
         } else {
-            this.functionCall = new FunctionNameFunctionCallModel(new FunctionName(this.functionCallConfig.getName()));
+            this.functionCall = BinaryData.fromObject(new FunctionName(this.functionCallConfig.getName()));
```

Comment:
Oh, I see, you have already adjusted the customization to account for the types used in the union.